src/utils/wallet.js

```diff
@@ -379,11 +379,18 @@ class Wallet {
     }
 
     async getBalance(accountId) {
-        let userAccountId = this.accountId;
-        if (accountId) {
-            userAccountId = accountId;
-        }
-        return await this.getAccount(userAccountId).getAccountBalance()
+        accountId = accountId || this.accountId
+
+        const account = this.getAccount(accountId)
+        const balance = await account.getAccountBalance()
+
+        // TODO: Should lockup contract balance be retrieved separately?
+        // TODO: Unhardcode lockup account name
+        const lockupAccountId = 'lockup-test.vg.betanet'
+        // TODO: Makes sense for a lockup contract to return whole state as JSON instead of method per property
```

Comment:
Do you suggest to modify the contract to add such a method? Right now rpc can only give state serialized in borsh

Reply:
@bowenwang1996 yeah, I suggest to modify the contract, otherwise it’s pretty inefficient to query the data